packages/core/builders/CalendarEvent/director.ts

```diff
@@ -0,0 +1,35 @@
+import { Booking } from "@prisma/client";
+
+import { CalendarEventBuilder } from "./builder";
+
+export class CalendarEventDirector {
+  private builder!: CalendarEventBuilder;
+  private existingBooking!: Partial<Booking>;
+  private cancellationReason!: string;
+
+  public setBuilder(builder: CalendarEventBuilder): void {
+    this.builder = builder;
+  }
+
+  public setExistingBooking(booking: Booking) {
+    this.existingBooking = booking;
+  }
+
+  public setCancellationReason(reason: string) {
+    this.cancellationReason = reason;
+  }
+
+  public async buildForRescheduleEmail(): Promise<void> {
+    if (this.existingBooking && this.existingBooking.eventTypeId && this.existingBooking.uid) {
+      await this.builder.buildEventObjectFromInnerClass(this.existingBooking.eventTypeId);
+      await this.builder.buildUsersFromInnerClass();
+      await this.builder.buildAttendeesList();
```

Comment:
This is not an async function:
```suggestion
      this.builder.buildAttendeesList();
```